### PR TITLE
Comparison made Mysql and Redis

### PR DESCRIPTION
**Pull request by igorlabzun**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.example</groupId>
+    <artifactId>FinalProjectModule_Four</artifactId>
+    <version>1.0-SNAPSHOT</version>
+
+    <properties>
+        <maven.compiler.source>18</maven.compiler.source>
```
Comment (Volodymyr95) [inline pom.xml:12]: use LTS version

File: src/main/java/com/labzun/domain/City.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    private Integer population;
+
+
+    public City(int i, String nameCity) {
+    }
+
+    public City() {
+
+    }
+
```
Comment (Volodymyr95) [inline src/main/java/com/labzun/domain/City.java:29]: formatting

File: src/main/java/com/labzun/domain/Main.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package com.labzun.domain;
+import com.labzun.redis.CityCountry;
+
+import java.util.List;
+
+public class Main {
+    public static void main(String[] args) {
+        MySessionFactory mySessionFactory = new MySessionFactory();
```
Comment (Volodymyr95) [inline src/main/java/com/labzun/domain/Main.java]: main method for app run, not for logic

File: src/main/java/com/labzun/domain/MySessionFactory.java [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    }
+
+    private RedisClient prepareRedisClient() {
+        RedisClient redisClient = RedisClient.create(RedisURI.create("localhost", 6379));
+        try (StatefulRedisConnection<String, String> connection = redisClient.connect()) {
+            System.out.println("\nConnected to Redis\n"+ connection);
+        }
+        return redisClient;
+    }
+    void pushToRedis(List<CityCountry> data) {
```
Comment (Volodymyr95) [inline src/main/java/com/labzun/domain/MySessionFactory.java]: why in MysqSessionFacttory class pushToRedis method?

File: src/main/java/com/labzun/domain/MySessionFactory.java [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.nonNull;
+
+public class MySessionFactory {
```
Comment (Volodymyr95) [inline src/main/java/com/labzun/domain/MySessionFactory.java]: what mean My?

File: src/main/java/com/labzun/domain/MySessionFactory.java [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+                String value = sync.get(String.valueOf(id));
+                try {
+                    mapper.readValue(value, CityCountry.class);
+                } catch (JsonProcessingException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+    }
+    void testMysqlData(List<Integer> ids) {
```
Comment (Volodymyr95) [inline src/main/java/com/labzun/domain/MySessionFactory.java]: is it test?

File: src/main/java/com/labzun/domain/MySessionFactory.java [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+        try (Session session = sessionFactory.getCurrentSession()) {
+            session.beginTransaction();
+            for (Integer id : ids) {
+                City city = cityDAO.getById(id);
+                Set<CountryLanguage> languages = city.getCountry().getLanguages();
+                System.out.println("Loaded "+ languages);
```
Comment (Volodymyr95) [inline src/main/java/com/labzun/domain/MySessionFactory.java]: use logger.info

**Review by Volodymyr95 — COMMENTED** [review]
code not formatted, missed logs

File: src/main/java/com/labzun/domain/ConnectionManager.java [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+    public void shutdown() {
+        if (nonNull(sessionFactory)) {
+            sessionFactory.close();
+        }
+        if (nonNull(redisClient)) {
+            redisClient.shutdown();
+        }
+    }
+
+    public List<City> fetchData(ConnectionManager connectionManager) {
```
Comment (Volodymyr95) [inline src/main/java/com/labzun/domain/ConnectionManager.java:84]: why u pass connection as argument?

**Review by Volodymyr95 — APPROVED** [review]

File: src/main/java/com/labzun/domain/ConnectionManager.java (hunk 8 above)
Comment (igorlabzun) [reply src/main/java/com/labzun/domain/ConnectionManager.java:84]: In the fetchData method, passing the ConnectionManager object as an argument allows its properties and methods, such as sessionFactory, cityDAO, and countryDAO, to be used within the method. Although sessionFactory, cityDAO, and other objects are already available in this class, explicitly passing them as an argument makes the method more independent of the object itself and improves its readability. It also makes the method less dependent on the state of the class and allows it to be tested separately.

**Pull request merged**